indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -93,7 +100,8 @@
   private final ServiceEmitter emitter;
 
   private final ReentrantLock giant = new ReentrantLock(true);
-  private final Condition managementMayBeNecessary = giant.newCondition();
+  @SuppressWarnings("MismatchedQueryAndUpdateOfCollection")
+  private final BlockingQueue<Object> managementMayBeNecessary = new ArrayBlockingQueue<>(8);
```

Comment:
Any reason why this is 8 instead of 1? I suppose it doesn't matter what the size is, since the queue is cleared in awaitManagementNanos. I'm just wondering if there's a reason I'm missing.

Reply:
No particular reason, I'll swap it to one.